### ISSUE TITLE
Edit rules: Jupyter Notebook crashes when too many rules

### ISSUE THREAD

**Issue by 80auline** [fivetran/great_expectations#4303]
**Describe the bug**
Edit rules: Jupyter Notebook crashes when too many rules.

I have a large table with ~50 columns. I generated rules using the historic data, it created ~650 rules.
When I try to open the Jupyter Notebook to edit the rules, it does not respond

**To Reproduce**
1. `great_expectations --v3-api suite edit my-table`
2. Jupyter notebook should be loading ... nothing happens and a message "Page

**Expected behavior**

- Clear information in the doc that specify the maximum number of columns or/and the maximum number of rules that can be processed by Jupyter Notebook.
- Jupyter Notebook does not crash and work normally 

**Environment (please complete the following information):**
 - Operating System: MacOS
 - Great Expectations Version: 0.14.8
 

<img width="1345" alt="Screenshot 2022-02-22 at 10 04 05" src="https://user-images.githubusercontent.com/11400917/156183850-41f3ed7a-f3d4-46ca-a0d9-b4cd0c182c09.png">

**Comment by talagluck** [comment]
Hi @80auline - it sounds like this is probably more due to system limitations than library limitations, but we will take a look and see if we can reproduce. In the meantime, a few follow-up questions:
- When you say rules, do you mean expectations?
- How much system memory do you have available when you try and run this?
- What profiler did you use that created the 650 Expectations?

As an aside, if you are on `0.14.8`, you no longer need to use the `--v3-api` flag when running from the CLI.

**Comment by joshua-stauffer** [comment]
Hi @80auline, thanks for raising this issue. Just wanted to chime in and say that I was able to reproduce something similar following these steps:
- create a wide and short csv (i used 72 cols, 2 rows, with arbitrary values)
- init a GE project, add a filesystem Pandas datasource
- run `great_expectations suite new` on the datasource, and use the UserConfigurableProfiler to profile the data (allowing the profiler to run on all columns), then save and close the notebook/jupyter server
- run `great_expectations suite edit <suite_name>` to reopen the juypter server

I didn't get the same exact error (running on Safari Version 15.3, looks like example above is on Chrome), but my notebook never loaded. I eventually got an alert saying that the page was using a lot of resources, and had to be restarted, but never got to the `suite edit` notebook.

I'm not immediately sure what the solution is here -- maybe an extra CLI step before opening the notebook, and if more than x amount of Expectations need to be populated, have a warning come up, and an extra confirm step before opening the notebook? But to your point @talagluck, I'm seeing this as system limitations and not library limitations.
